src/test/kotlin/org/rust/ide/inspections/lints/RsUnreachableCodeInspectionTest.kt

```diff
@@ -229,4 +229,62 @@ class RsUnreachableCodeInspectionTest : RsInspectionsTestBase(RsUnreachableCodeI
             <warning descr="Unreachable code">()</warning>
         }
     """)
+
+    fun `test incomplete match expr reachable`() = checkByText("""
+        fn main() {
+            let x = match 3 {
+
+            };
+            return;
+        }
+    """)
+
+    fun `test incomplete match stmt reachable`() = checkByText("""
+        fn main() {
+            match Some(true) {
```

Comment:
Let's avoid using `Option` type (here and below) because it requires stdlib loading (see tests with `@ProjectDescriptor(WithStdlibRustProjectDescriptor::class)` annotations). It's convenient to use a local enum declaration like this:
```Rust
enum E { A, B }
fn foo(e: E) {
    match e {
        E::A => {}
    }
}
```